flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/CollectStreamTableSink.java

```diff
@@ -43,21 +43,32 @@
 	private final TypeSerializer<Tuple2<Boolean, Row>> serializer;
 	private final TableSchema tableSchema;
 
-	public CollectStreamTableSink(InetAddress targetAddress, int targetPort, TypeSerializer<Tuple2<Boolean, Row>> serializer, TableSchema tableSchema) {
+	public CollectStreamTableSink(
+			InetAddress targetAddress,
+			int targetPort,
+			TypeSerializer<Tuple2<Boolean, Row>> serializer,
+			TableSchema tableSchema) {
 		this.targetAddress = targetAddress;
 		this.targetPort = targetPort;
 		this.serializer = serializer;
 		this.tableSchema = TableSchemaUtils.checkNoGeneratedColumns(tableSchema);
```

Comment:
Can we override `getTableSchema` directly:
```
    @Override
	public TableSchema getTableSchema() {
		// We use the old type system here to be consistent with getRecordType.
		return new TableSchema(tableSchema.getFieldNames(), tableSchema.getFieldTypes());
	}
```
I think `getFieldNames` and `getFieldTypes` is not so intuitive.

Reply:
I could not decide on those two options myself. 

I went for the `getFieldNames` and `getFieldTypes` methods because I thought it will be easier to track it back when fixing the `Upsert/RetractTableSink` interfaces. I thought it will be more visible that we need to fix it here as well when removing those methods. I am afraid the old type system in `TableSchema` might stick around for a bit longer than these methods. Moreover it is also how we do it e.g. in Elasticsearch retract stream.

But I am fine with switching over to `TableSchema` if you think it would be better that way.